### PR TITLE
Completely removed Alamofire in favor of simple networking calls.

### PR DESCRIPTION
**Pull request by macblazer**
This commit removes the Alamofire dependency with no other changes.

This fixes one part of issue #48.

### CODE REVIEW

File: Sources/PactVerificationService.swift [hunk 1]
```diff
@@ -91,30 +108,67 @@ open class PactVerificationService {
 
   fileprivate func verifyInteractions() -> Future<String, NSError> {
     let promise = Promise<String, NSError>()
-    Alamofire.request(Router.verify())
-    .validate()
-    .responseString { response in self.requestHandler(promise)(response) }
+
+    do {
```
Comment (andrewspinks) [inline Sources/PactVerificationService.swift]: Is it possible to create a function to consolidate this error handling code?

File: Sources/PactVerificationService.swift [hunk 2]
```diff
@@ -45,20 +44,32 @@ open class PactVerificationService {
...
       default:
         return urlRequest
       }
     }
+
+        private func jsonEncode(_ request: URLRequest, with parameters: [String: Any]) throws -> URLRequest {
+            var urlRequest = request
+            let data = try JSONSerialization.data(withJSONObject: parameters, options: [])
+
+            if urlRequest.value(forHTTPHeaderField: "Content-Type") == nil {
```
Comment (andrewspinks) [inline Sources/PactVerificationService.swift]: Is this if statement necessary? I think its fine to always set the content type.

**Comment by andrewspinks** [comment]
@macblazer Thanks heaps for the clean PR. I'll try and take a closer look ASAP.

File: Tests/PactConsumerSwiftTests/AnimalServiceClient.swift [hunk 3]
```diff
@@ -9,128 +8,162 @@ public struct Animal {
...
+            type: alligator["type"] as! String,
+            dob: alligator["dateOfBirth"] as? String,
+            legs: alligator["legs"] as? Int))
+        }
+        success(animals)
+      }
+    }
+  }
+
+  private func performRequest(_ urlString: String, headers: [String: String]? = nil, method: String = "get", parameters: [String: String]? = nil, isJsonResponse: Bool = true, completionHandler: @escaping (_ jsonObject: Any?, _ error: NSError?) -> Void) {
```
Comment (surpher) [inline Tests/PactConsumerSwiftTests/AnimalServiceClient.swift]: There are so many arguments in this function declaration in one line, please consider the following suggestion for readability:
```
performRequest(
  _ urlString: String, 
  headers: [String: String]? = nil, 
  method: String = "get", 
  parameters: [String: String]? = nil, 
  isJsonResponse: Bool = true, 
  completionHandler: @escaping (_ jsonObject: Any?, _ error: NSError?) -> Void
) {
...
```

Am also a bit surprised _SwiftLint_ has not picked up this `line_length` violation.

File: Tests/PactConsumerSwiftTests/AnimalServiceClient.swift [hunk 4]
```diff
@@ -9,128 +8,162 @@ public struct Animal {
...
-                  })
-                } else {
-                  failure(result.result.value as? NSError)
-                }
-              }
-            }
+    self.performRequest("\(baseUrl)/alligators") { jsonObject, nsError in
+      if let jsonArray = jsonObject as? Array<[String: AnyObject]> {
+        success(jsonArray.map { animal -> Animal in
+          return Animal(
```
Comment (surpher) [inline Tests/PactConsumerSwiftTests/AnimalServiceClient.swift]: I understand this is a test, though would suggest setting up an `Animal` object that conforms to `Codable` protocol. Type casting would not be required anymore.

**Review by surpher — COMMENTED** [review]
Looks good, thanks for contributing! 
There are a few minor improvements I would suggest as per comments I left.

File: Tests/PactConsumerSwiftTests/AnimalServiceClient.swift (hunk 4 above)
Comment (macblazer) [reply Tests/PactConsumerSwiftTests/AnimalServiceClient.swift]: I agree

File: Sources/PactVerificationService.swift (hunk 1 above)
Comment (macblazer) [reply Sources/PactVerificationService.swift]: Probably.  I will see if I can make that simpler.

**Review by andrewspinks — APPROVED** [review]
@macblazer Thanks heaps for the PR! A couple of little comments

**Comment by andrewspinks** [comment]
@macblazer LGTM, I'll merge it unless you have any more changes you want to make?

**Comment by macblazer** [comment]
I have no more changes to make.  Thanks for getting this merged.

**Pull request merged**